sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/implementation/util/BlobUserAgentModificationPolicy.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.implementation.util;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpPipelinePosition;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * This policy modifies the blob user agent string for clients created in packages that are dependencies of blob.
+ * It transforms a User Agent String as follows
+ * UAbefore: "azsdk-java-azure-storage-blob/12.11.0-beta.2 (11.0.6; Windows 10; 10.0)"
+ * UAafter: "azsdk-java-azure-storage-blob/12.11.0-beta.2 azsdk-java-azure-storage-blob-batch/12.8.0-beta.2 (11.0.6; Windows 10; 10.0) "
+ */
+public class BlobUserAgentModificationPolicy implements HttpPipelinePolicy {
+
+    private final String clientName;
+    private final String clientVersion;
+
+    private static final String USER_AGENT = "User-Agent";
+    private static final String REGEX = "azsdk-java-azure-storage-blob/12\\.\\d{1,2}\\.\\d{1,2}(?:-beta\\.\\d{1,2})?";
```

Comment:
To cover all cases we want to cover,

- User-Agent with prepended custom application ID: `myapplicationid azure-java-azure-storage-blob/12.10.0`
- User-Agent with appended OS and JVM info: `azure-java-azure-storage-blob/12.10.0 (Java/1.11; Windows)`
- User-Agent with both prepended custom application ID and appended OS and JVM info: `myapplicationid azure-java-azure-storage-blob/12.10.0 (Java/1.11; Windows)`
- User-Agent with no prepended or appended information: `azure-java-azure-storage-blob/12.10.0`

I would update the regex pattern to the following:

`(.*? )?(azure-java-azure-storage-blob/12\\.\\d{1,2}\\.\\d{1,2}(?:-beta\\.\\d{1,2})?)( .*?)?`

This should match the entire User-Agent if it is using the standard format, it should contain 4 capture groups as follows:

Group 0: The entire string
Group 1: An optional application ID
Group 2: The standard Storage Blobs User-Agent
Group 3: An optional OS and JVM info

This can then be used by the replacement logic as follows:

```java
if (matcher.matches()) {
    StringBuilder newUserAgentBuilder = new StringBuilder(matcher.group(1))
        .append(matcher.group(2))
        .append(/* logic to add additional Storage library User-Agent */)
        .append(matcher.group(3));
    
     String newUserAgent = newUserAgentBuilder.toString();
```